kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/dhcp/DhcpServerConfigIP4.java

```diff
@@ -51,12 +56,117 @@
+     * @param rangeEnd
+     *            the network ending address to issue to DHCP clients
+     * @param passDns
+     *            whether or not to pass DNS to DHCP clients
+     * @param dnsServers
+     *            the DNS servers that will get passed to DHCP clients if passDns is true
+     */
+	private DhcpServerConfigIP4(boolean enabled, String interfaceName, IP4Address subnet, IP4Address routerAddress,
+            IP4Address subnetMask, int defaultLeaseTime, int maximumLeaseTime, short prefix, IP4Address rangeStart,
+            IP4Address rangeEnd, boolean passDns, List<IP4Address> dnsServers) {
 
+    	super(interfaceName, enabled, subnet, routerAddress, subnetMask, defaultLeaseTime, maximumLeaseTime, prefix,
+                rangeStart, rangeEnd, passDns, dnsServers);
     }
+	
+	/**
+     * A method that invokes the Constructor after checking that DHCP pool IP addresses are in the subnet.
+     *
+     * @param interfaceName
+     *            the interface name associated with the DhcpServerConfig
+     * @param enabled
+     *            the status of the DhcpServer as a boolean
+     * @param subnet
+     *            the subnet of the DhcpServerConfig
+     * @param routerAddress
+     *            the router IPAddress
+     * @param subnetMask
+     *            the subnet mask of the DhcpServerConfig
+     * @param defaultLeaseTime
+     *            the default lease time to issue to DHCP clients
+     * @param maximumLeaseTime
+     *            the maximum lease time to issue to DHCP clients
+     * @param prefix
+     *            the network prefix associated with the DhcpServerConfig
+     * @param rangeStart
+     *            the network starting address to issue to DHCP clients
+     * @param rangeEnd
+     *            the network ending address to issue to DHCP clients
+     * @param passDns
+     *            whether or not to pass DNS to DHCP clients
+     * @param dnsServers
+     *            the DNS servers that will get passed to DHCP clients if passDns is true
+     */
+    public static DhcpServerConfigIP4 newDhcpServerConfigIP4(String interfaceName, boolean enabled, IP4Address subnet, IP4Address routerAddress,
+            IP4Address subnetMask, int defaultLeaseTime, int maximumLeaseTime, short prefix, IP4Address rangeStart,
+            IP4Address rangeEnd, boolean passDns, List<IP4Address> dnsServers)  throws KuraException {
+		
+		DhcpServerConfigIP4 retVal = null;
+		if (isIpAddressInSubnet(rangeStart.getHostAddress(), subnet.getHostAddress(), subnetMask.getHostAddress())
+				&& isIpAddressInSubnet(rangeEnd.getHostAddress(), subnet.getHostAddress(),
+						subnetMask.getHostAddress())) {
+
+			retVal = new DhcpServerConfigIP4(enabled, interfaceName, subnet, routerAddress, subnetMask, defaultLeaseTime, maximumLeaseTime, prefix,
+	                rangeStart, rangeEnd, passDns, dnsServers);
+		} 
+
+		return retVal;
+	}
+    
+    private static int inet4address2int(Inet4Address inet4addr) {
```

Comment:
Yes, I've seen it, but we can't use NetworkUtil from classes in the org.eclipse.kura.api. This is why I added the inet4address2int() method to the DhcpServerConfigIP4 class. The reason I wanted to address the issue on this leves is to prevent Kura users from creating the DhcpServerConfigIP4 objects with invalid range.